python/sglang/srt/server.py

```diff
@@ -123,31 +123,91 @@ async def flush_cache():
     )
 
 
-async def detokenize_logprob_tokens(token_logprobs):
-    token_ids = [tid for tid, _ in token_logprobs]
+async def detokenize_logprob_tokens(token_logprobs, decode_to_text):
+    if not decode_to_text:
+        return [(logprob, token_id, None) for logprob, token_id in token_logprobs]
+
+    token_ids = [tid for _, tid in token_logprobs]
     token_texts = await tokenizer_manager.detokenize(DetokenizeReqInput(token_ids))
-    return [(text, logprob) for text, (_, logprob) in zip(token_texts, token_logprobs)]
+    return [
+        (logprob, token_id, token_text)
+        for (logprob, token_id), token_text, in zip(token_logprobs, token_texts)
+    ]
+
+
+async def detokenize_top_logprobs_tokens(top_logprobs, decode_to_text):
+    for i, t in enumerate(top_logprobs):
+        if top_logprobs[i] is not None:
+            top_logprobs[i] = await detokenize_logprob_tokens(t, decode_to_text)
```

Comment:
nit: Can we use async loop?

Reply:
Oh, the `top_logprobs` here is not an async iterable object, it is a regular list, so we cannot use `async for` here.